packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -34,6 +34,26 @@ jobject wrap_pointer(JNIEnv* jenv, jlong pointer, jboolean managed = false) {
                            managed);
 }
 
+inline jboolean jni_check_exception(JNIEnv *jenv = get_env(), bool registrable_callback_error = false) {
+    if (jenv->ExceptionCheck()) {
+        jthrowable exception = jenv->ExceptionOccurred();
+        jenv->ExceptionClear();
+        // setting the user code error is only propagated on certain callbacks.
+        if (registrable_callback_error)
```

Comment:
I think we should link to https://github.com/realm/realm-kotlin/issues/665 to throw the exception if we don't want to rely on the `registrable_callback_error` mechanism, otherwise this method calls acts as catch and swallow ...